packages/beacon-node/src/network/subnets/attnetsService.ts

```diff
@@ -172,7 +172,16 @@ export class AttnetsService implements IAttnetsService {
    */
   private unsubscribeExpiredRandomSubnets(slot: Slot): void {
     const expired = this.subscriptionsRandom.getExpired(slot);
-    // TODO: Optimization: If we have to be subcribed to all subnets, no need to unsubscribe. Just extend the timeout
+    const currentSlot = this.chain.clock.currentSlot;
+
+    if (this.knownValidators.size * RANDOM_SUBNETS_PER_VALIDATOR >= ATTESTATION_SUBNET_COUNT) {
+      // Optimization: If we have to be subcribed to all subnets, no need to unsubscribe. Just extend the timeout
+      for (const subnet of expired) {
+        this.subscriptionsRandom.request({subnet, toSlot: randomSubscriptionSlotLen() + currentSlot});
+      }
+      return;
+    }
```

Comment:
Does this also cover the --subscribe-all-subnets flag

Reply:
@dapplion  it's already covered https://github.com/ChainSafe/lodestar/blob/1fde79dad53c91c6df05efb04b12a09bd1275d00/packages/beacon-node/src/network/subnets/attnetsService.ts#L276